marathon_acme/tests/test_clients.py

```diff
@@ -1219,3 +1219,130 @@ def test_not_found(self):
         self.stub_client.flush()
 
         assert_that(d, succeeded(Is(None)))
+
+    def test_read_kv2(self):
+        """
+        When data is read from the key/value version 2 API, the response is
+        returned.
+        """
+        d = self.client.read_kv2('hello')
+
+        request_d = self.requests.get()
+        assert_that(request_d, succeeded(MatchesAll(
+            HasRequestProperties(method='GET', url='/v1/secret/data/hello',
+                                 query={}),
+            MatchesStructure(
+                requestHeaders=HasHeader('X-Vault-Token', [self.token]))
+        )))
+
+        # Write the response
+        dummy_response = {
+            "request_id": "08b0ba90-b6e4-afab-de6a-d2fbf8f480b3",
+            "lease_id": "",
+            "renewable": False,
+            "lease_duration": 0,
+            "data": {
+                "data": {"foo": "world"},
+                "metadata": {
+                    "created_time": "2018-09-05T12:49:52.722404Z",
+                    "deletion_time": "",
+                    "destroyed": False,
+                    "version": 1
+                }
+            },
+            "wrap_info": None,
+            "warnings": None,
+            "auth": None
+        }
+
+        request = request_d.result
+        request.setResponseCode(200)
+        write_request_json(request, dummy_response)
+        request.finish()
+        self.stub_client.flush()
+
+        # Response should be returned
+        assert_that(d, succeeded(Equals(dummy_response)))
+
+    def test_read_kv2_with_version(self):
+        """
+        When data is read from the key/value version 2 API and a version is
+        specified, the version parameter is sent.
+        """
+        self.client.read_kv2('hello', version=1)
+
+        request_d = self.requests.get()
+        assert_that(request_d, succeeded(MatchesAll(
+            HasRequestProperties(method='GET', url='/v1/secret/data/hello',
+                                 query={'version': ['1']}),
+            MatchesStructure(
+                requestHeaders=HasHeader('X-Vault-Token', [self.token]))
+        )))
+
+        # Just checking the version parameter is sent
```

Comment:
Should this comment be above the `assert_that()` call? At the moment it looks like it's explaining code that isn't there.

Reply:
I guess it was meant to be in place of the code that might follow to assert on the returned value...but I see how that could be misunderstood. I think the docstrings for the tests are fine and I'll just remove these.